add endpoint GET /api/articles/:article_id/comments

robert-matthew-brooks opened the pull request:
ps one of the test has to skipped as it will fail without the endpoint from the previous pull request

robert-matthew-brooks committed:
add endpoint GET /api/articles/:article_id/comments

ps one of the test has to skipped as it will fail without the endpoint from the previous pull request

robert-matthew-brooks committed:
Merge branch 'main' of https://github.com/robert-matthew-brooks/be-nc-news into be-nc-news-6

Slyose commented on models/comments.models.js:
```diff
@@ -0,0 +1,48 @@
+const db = require('../db/connection.js');
+
+class CommentsModels {
```
You may or may not have heard this from other mentors but I'll mention it now - declaring all of your functions as a static method on an object is an unnecessary step that is still functionally identical to just declaring the functions on their own with a const. The only difference is that you now need to use for example `CommentsModels.getComments()` instead of just `getComments` by itself, and you don't get any of the benefits of OOP. This object is essentially acting as a wrapper around your functions that could just be exported as-is. 

Express is designed to work functionally, and you will end up passing a function into everything anyway, which is why this is an unusual approach to building an express server.

Slyose reviewed (commented):
I left a comment about your OOP style employed in this app which I'd like you to take a look at. Apart from that, take out your console log then feel free to merge!

robert-matthew-brooks committed:
refactor based on feedback

no new functionality, just refactoring
controller functions now wrapped in an object instead of a class
removed hardcoded tests that were too brittle and unneccessary
combined multiple tests (status code, array length, object key/value layout) into single tests
